watertap/property_models/unit_specific/activated_sludge/asm3_properties.py

```diff
@@ -0,0 +1,595 @@
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+"""
+Thermophysical property package to be used in conjunction with ASM3 reactions.
+"""
+
+# Import Pyomo libraries
+import pyomo.environ as pyo
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    MaterialFlowBasis,
+    StateBlockData,
+    StateBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    LiquidPhase,
+    Component,
+    Solute,
+    Solvent,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import fix_state_vars, revert_state_vars
+from idaes.core.scaling import CustomScalerBase
+import idaes.logger as idaeslog
+from idaes.core.base.property_base import PhysicalParameterBlock
+import idaes.core.util.scaling as iscale
+
+# Some more information about this module
+__author__ = "Chenyu Wang"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("ASM3ParameterBlock")
+class ASM3ParameterData(PhysicalParameterBlock):
+    """
+    Property Parameter Block Class
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super().build()
+
+        self._state_block_class = ASM3StateBlock
+
+        # Add Phase objects
+        self.Liq = LiquidPhase()
+
+        # Add Component objects
+        self.H2O = Solvent()
+
+        # Soluble Components
+        self.S_O = Solute(doc="Dissolved Oxygen, S_O")
+        self.S_I = Solute(doc="Inert soluble organic material, S_I")
+        self.S_S = Solute(doc="Readily biodegradable substrates, S_S")
+        self.S_NH4 = Solute(
+            doc="Ammonium plus ammonia nitrogen (NH4+ + NH3 nitrogen), S_NH4"
+        )
+        self.S_N2 = Solute(doc="Dinitrogen (N2), S_N2")
+        self.S_NOX = Solute(
+            doc="Nitrate plus nitrite nitrogen (NO3- + NO2- nitrogen), S_NOX"
+        )
+        self.S_ALK = Component(doc="Alkalinity (HCO3-), S_ALK")
+
+        # Particulate Components
+        self.X_I = Solute(doc="Inert particulate organic matter, X_I")
+        self.X_S = Solute(doc="Slowly biodegradable substrates, X_S")
+        self.X_H = Solute(doc="Heterotrophic organisms, X_H")
+        self.X_STO = Solute(
+            doc="A cell internal storage product of heterotrophic organisms, X_STO"
+        )
+        self.X_A = Solute(doc="Nitrifying organisms, X_A")
```

Comment:
Is this docstring accurate? This also loosely ties to the comment below

Reply:
Yeah, X_A is Nitrifying organisms